### PR TITLE
Phar release - fix readme generation

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2805

### CODE REVIEW

File: src/Console/Command/HelpCommand.php [hunk 1]
```diff
@@ -376,6 +380,21 @@ protected function initialize(InputInterface $input, OutputInterface $output)
...
+     * @return string|null
+     */
+    private static function getChangeLogFile()
+    {
+        $changelogFile = __DIR__.'/../../../CHANGELOG.md';
+        if (!is_file($changelogFile)) {
+            return null;
+        }
+
+        $changelogFile = realpath($changelogFile);
```
Comment (keradus) [inline src/Console/Command/HelpCommand.php]: why realpath is needed ?

Comment (GrahamCampbell) [inline src/Console/Command/HelpCommand.php]: In order to tell if it exists. It gives `false` if the path doesn't resolve, which is used on the next line.

Comment (keradus) [inline src/Console/Command/HelpCommand.php]: we already checked that it exists 4 lines before, and we don't care about linkshere...

Comment (GrahamCampbell) [inline src/Console/Command/HelpCommand.php]: the behaviour of realpath inside a phar is different

Comment (SpacePossum) [inline src/Console/Command/HelpCommand.php]: That is what I want to try out here as the file should always exists (both phar and non phar). For non-phar it should always be readable, for phar it is not (that is reported). Using `realpath` should result OK for non-phar and `false` for phar so the fallback would be used. This way we won't hide a failed readable check for non-phar.
Lets see what happens ;)

Comment (SpacePossum) [inline src/Console/Command/HelpCommand.php]: btw. how can I create a phar locally to test?

Comment (GrahamCampbell) [inline src/Console/Command/HelpCommand.php]: I don't think so. It just returns `false` without error. :)

Comment (keradus) [inline src/Console/Command/HelpCommand.php]: > That is what I want to try out here as the file should always exists

no, it's not (and it shouldn't)

Comment (SpacePossum) [inline src/Console/Command/HelpCommand.php]: all green without it, but I didn't test the phar because I don't know how

**Comment by keradus** [comment]
works !

```
D:\GIT\fork\PHP-CS-Fixer ((detached from SpacePossum/2_2_missing_changelog_handling))
λ php php-cs-fixer readme | head -4
PHP Coding Standards Fixer
==========================

The PHP Coding Standards Fixer tool fixes *most* issues in your code when you

D:\GIT\fork\PHP-CS-Fixer ((detached from SpacePossum/2_2_missing_changelog_handling))
λ php php-cs-fixer.phar readme | head -4
PHP Coding Standards Fixer
==========================

The PHP Coding Standards Fixer tool fixes *most* issues in your code when you
```

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**